scripts/zones/Maze_of_Shakhrami/mobs/Leech_King.lua

```diff
@@ -0,0 +1,23 @@
+-----------------------------------
+-- Area: Maze of Shakhrami
+--  MOB: Leech King (NM)
+-----------------------------------
+
+require("scripts/zones/Maze_of_Shakhrami/MobIDs");
+
+-----------------------------------
+-- onMobDeath
+-----------------------------------
+
+function onMobDeath(mob,killer,ally)
+    local whichNM = math.random(0,19);
+
+    if (whichNM < 10) then
+        SetRespawnTime(Argus, 64800, 108000); -- 18-30 hours
+        DeterMob(Leech_King, true);
+    else
+        SetRespawnTime(Leech_King, 64800, 108000); -- 18-30 hours
+        DeterMob(Argus, true);
```

Comment:
@Hozu took a look at it, but this would make the logic more complex
the problem is, that the respawn time changes (from 0-12h to 18-30h) after the first kill of each NM, so you get more checking + 2x deter/1xsetRespawn instead of no checking and 1x setRespawn + 1x deter